listenbrainz_spark/recommendations/tests/test_dataframe.py

```diff
@@ -0,0 +1,152 @@
+import uuid
+import unittest
+from datetime import datetime
+
+import listenbrainz_spark
+from listenbrainz_spark.tests import SparkTestCase
+import listenbrainz_spark.recommendations.create_dataframes as dataframe
+from listenbrainz_spark import schema, utils, config, path, hdfs_connection
+
+from pyspark.sql import Row
+
+# path used in between test functions of this class
+LISTENS_PATH = '/test/listens/' # HDFS path to listens
+MAPPING_PATH = '/test/mapping.parquet'
+MAPPED_LISTENS_PATH = '/test/mapped_listens.parquet'
+
+DATE = datetime.utcnow()
+
+class CreateDataframeTestclass(SparkTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.upload_test_listen_to_HDFS()
+        cls.upload_test_mapping_to_HDFS()
+        cls.upload_test_mapped_listens_to_HDFS()
+
+    @classmethod
+    def tearDownClass(cls):
+        super().delete_dir()
+        super().tearDownClass()
+
+    @classmethod
+    def upload_test_listen_to_HDFS(cls):
+        month, year = DATE.strftime('%m'), DATE.strftime('%Y')
+
+        test_listen = {
+            "user_name": "vansika", "artist_msid": "a36d6fc9-49d0-4789-a7dd-a2b72369ca45",
+            "artist_name": "Less Than Jake", "artist_mbids": [], "release_mbid": "", "track_name": "Al's War",
+            "recording_msid": "cb6985cd-cc71-4d59-b4fb-2e72796af741", "tags": [], "listened_at": DATE
+        }
+
+        test_listens_df = utils.create_dataframe([schema.convert_to_spark_json(test_listen)], schema.listen_schema)
+        utils.save_parquet(test_listens_df, LISTENS_PATH + '{}/{}.parquet'.format(year, month))
+
+    @classmethod
+    def upload_test_mapping_to_HDFS(cls):
+        test_mapping = {"msb_recording_msid":"cb6985cd-cc71-4d59-b4fb-2e72796af741"
+            ,"mb_recording_gid":"3acb406f-c716-45f8-a8bd-96ca3939c2e5","msb_artist_msid":"a36d6fc9-49d0-4789-a7dd-a2b72369ca45",
+            "mb_artist_gids":["181c4177-f33a-441d-b15d-910acaf18b07"],"mb_artist_credit_id":2157963}
+
+        test_mapping_df = utils.create_dataframe([schema.convert_mapping_to_row(test_mapping)], schema.mapping_schema)
+        utils.save_parquet(test_mapping_df, MAPPING_PATH)
+
+    @classmethod
+    def upload_test_mapped_listens_to_HDFS(cls):
+        partial_listen_df = dataframe.get_listens_for_training_model_window(DATE, DATE, {}, LISTENS_PATH)
+        mapping_df = utils.read_files_from_HDFS(MAPPING_PATH)
+
+        mapped_df = dataframe.get_mapped_artist_and_recording_mbids(partial_listen_df, mapping_df)
+        utils.save_parquet(mapped_df, MAPPED_LISTENS_PATH)
+
+    def test_get_dates_to_train_data(self):
+        to_date, from_date = dataframe.get_dates_to_train_data()
+        self.assertLessEqual(from_date, to_date)
```

Comment:
Might be a good idea to assert that the difference between the two dates is what we expect

Reply:
Asserting the diff is not possible here IMO because after we go back x steps from to_date to get the from_date, we shift the from_date to the first of the month (let us say y steps). We know x but not y, and the diff can only be checked using from_date + x + y = to_date. 
I will try to find a better test though :)

Reply:
The current implementation looks good to me!